### PR TITLE
Migrate `BaseSection` component to MDX

### PR DESCRIPTION
**Pull request by mcellier**
## Description

Migrate `BaseSection` component to MDX

## How it was tested

Storybook + Edit options page on kairos

### CODE REVIEW

File: src/layout/section/baseSection/BaseSection.style.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import styled from 'styled-components'
+
+import { componentSizes, horizontalSpace, responsiveBreakpoints } from '../../../_utils/branding'
+
+export const StyledBaseSection = styled.div`
```
Comment (gabrielagabriel) [inline src/layout/section/baseSection/BaseSection.style.tsx:5]: should base "section" be a `section` tag?

File: src/layout/section/baseSection/BaseSection.style.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    padding-left: ${horizontalSpace.global};
+    padding-right: ${horizontalSpace.global};
+  }
+
+  & .section-content--noHorizontalSpacing {
+    padding-left: 0;
+    padding-right: 0;
+  }
+
+  @media (${responsiveBreakpoints.isMediaLarge}) {
```
Comment (gabrielagabriel) [inline src/layout/section/baseSection/BaseSection.style.tsx:16]: nit breakpoints inside classes to clear reading on what is changing on each part of the component

**Review by gabrielagabriel — APPROVED** [review]
💯

File: src/layout/section/baseSection/BaseSection.style.tsx (hunk 1 above)
Comment (mcellier) [reply src/layout/section/baseSection/BaseSection.style.tsx:5]: Not in the scope of this PR I guess, but I don't have a strong feeling about this

**Pull request merged**